prbt_hardware_support/test/unittests/unittest_adapter_sto.cpp

```diff
@@ -0,0 +1,827 @@
+/*
```

Comment:
Doing `stress --cpu 8 --io 4 --vm 2` and `for i in {0..100}; do catkin_make run_tests_prbt_hardware_support_gtest_unittest_adapter_sto; done`

results sometimes in stuck testcase 
- AdapterStoTest.testSpamEnablePlusStop

Sadly I cannot reproduce with Level DEBUG showing the state transition, but this indicates some timing issue where I would suspect the spaming to be involved...

Reply:
I suggest dropping the spam. I figure you needed it to achieve full coverage?

Reply:
Simply changing the debug level (log4j.logger.ros.prbt_hardware_support.STOStateMachine=DEBUG) has a huge impact on the test result. This unittest should be **re-written in a purely deterministic way**.

Reply:
The problems seem to be resolved after removing the is_executing calls. I would not re-write the tests since I really focused on being deterministic.
However if you can figure a concrete part of the test that is not deterministic, please show me.

Reply:
Having tests fail here and there suggested some kind of non deterministic behaviour. I will have a closer look at it.

Reply:
At 34aa704b5e66fc2c25dd1c033cbeea8cc7f6ca93

I ran `for i in {0..100}; do catkin_make run_tests_prbt_hardware_support_gtest_unittest_adapter_sto; done`
an immediatly got stuck with:
- AdapterStoTest.testSpamEnablePlusStop

Reply:
Also

- AdapterStoTest.testUnholdFail get stuck